### PR TITLE
Reimplement scripting over plain UDP

### PR DESCRIPTION
**Pull request by wwylele**
The deadline I gave was over, so now I nuked libzmq dependency. 

The `udp_server.cpp` file wasn't written from scratch but adapted from the original `zmq_server.cpp`. It might contain some unusual code style from me.

Also removed the option in cmake and always enable scripting because now it costs basically nothing.

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/4609)

### CODE REVIEW

File: src/core/rpc/server.cpp [hunk 1]
```diff
@@ -1,27 +1,30 @@
...
-    const auto callback = [this](std::unique_ptr<RPC::Packet> new_request) {
+    const auto callback = [this](std::unique_ptr<Packet> new_request) {
         NewRequestCallback(std::move(new_request));
     };
 
     try {
-        zmq_server = std::make_unique<ZMQServer>(callback);
+        udp_server = std::make_unique<UDPServer>(callback);
     } catch (...) {
         LOG_ERROR(RPC_Server, "Error starting ZeroMQ server");
```
Comment (tgsm) [inline src/core/rpc/server.cpp]:
```suggestion
         LOG_ERROR(RPC_Server, "Error starting UDP server");
```

File: src/core/rpc/udp_server.cpp [hunk 2]
```diff
@@ -0,0 +1,94 @@
+#include <thread>
```
Comment (zhaowenlan1779) [inline src/core/rpc/udp_server.cpp]:
```suggestion
// Copyright 2019 Citra Emulator Project
// Licensed under GPLv2 or any later version
// Refer to the license.txt file included.

#include <thread>
```

File: src/core/rpc/udp_server.cpp [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+                u8* data = request_buffer.data() + MIN_PACKET_SIZE;
+                std::function<void(Packet&)> send_reply_callback =
+                    std::bind(&Impl::SendReply, this, remote_endpoint, std::placeholders::_1);
+                std::unique_ptr<Packet> new_packet =
+                    std::make_unique<Packet>(header, data, send_reply_callback);
+
+                // Send the request to the upper layer for handling
+                new_request_callback(std::move(new_packet));
+            }
+        }
```
Comment (zhaowenlan1779) [inline src/core/rpc/udp_server.cpp]: should an error be displayed when the size/header is not correct?

File: src/core/rpc/udp_server.cpp [hunk 4]
```diff
@@ -0,0 +1,100 @@
+// Copyright 2019 Citra Emulator Project
+// Licensed under GPLv2 or any later version
+// Refer to the license.txt file included.
+
+#include <thread>
+#include <boost/asio.hpp>
+#include "common/common_types.h"
+#include "core/core.h"
```
[the remark this answers is not in the document]
Comment (wwylele) [reply src/core/rpc/udp_server.cpp]: No where. Leftover from old code.

**Pull request merged**